### PR TITLE
Integrating FlashDecode_v2 to llama3 t3k

### PR DESCRIPTION
**Pull request by kevin-mii**

### CODE REVIEW

**Comment by mikevin920** [comment]
T3K Unit:
Post-commit:

**Comment by cglagovichTT** [comment]
New pipelines:
https://github.com/tenstorrent/tt-metal/actions/runs/10014023607

**Comment by cglagovichTT** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10014033622

**Comment by cglagovichTT** [comment]
t3k frequent: https://github.com/tenstorrent/tt-metal/actions/runs/10041223682

**Comment by avoraTT** [comment]
t3k demo with new ground truth: https://github.com/tenstorrent/tt-metal/actions/runs/10045664115

File: tests/scripts/t3000/run_t3000_demo_tests.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ run_t3000_llama3_70b_tests() {
   echo "LOG_METAL: Running run_t3000_llama3_70b_tests"
 
   # Llama3 70B demo (output verification)
-  env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/t3000/llama3_70b/demo/demo.py::test_LlamaModel_demo[wormhole_b0-True-check_enabled-greedy-tt-70b-T3000-80L-decode_only-text_completion-llama3] --timeout=900
+  env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/t3000/llama3_70b/demo/demo.py::test_LlamaModel_demo[wormhole_b0-True-short_context-check_enabled-greedy-tt-70b-T3000-80L-decode_only-text_completion-llama3] --timeout=900
```
Comment (ttmchiou) [inline tests/scripts/t3000/run_t3000_demo_tests.sh:34]: there's a movement to cut down on directly invoking specific pytests in test files and to invoke the .py file directly (without **.py::[])

can we follow this convention here?
Looking at demo.py there doesn't seem to be other tests so I think this is a trivial change.
Though the README.md will need to be updated as well to match invoking this pytest in particular

Comment (cglagovichTT) [reply tests/scripts/t3000/run_t3000_demo_tests.sh:34]: We need to leave all those options available so we/users can run the demo with a variety of args.

Can we open a separate issue so we can create a `demo_t3000.py` test file just like we've done for all the other pytests?

Comment (ttmchiou) [reply tests/scripts/t3000/run_t3000_demo_tests.sh:34]: sounds good

File: models/demos/t3000/llama2_70b/tt/llama_common.py [hunk 2]
```diff
@@ -415,3 +415,28 @@ def check_kv_cache(pt_cache_all, tt_cache_all, generation_start_pos, generation_
             logger.warning(f"KV Cache Failed! PCC value is lower than {pcc}")
             test_passed = False
     return test_passed
+
+
+def get_flash_decode_chunk_size(s):
```
Comment (caixunshiren) [inline models/demos/t3000/llama2_70b/tt/llama_common.py:420]: I believe flash decode v2 does not need this function anymore

File: tests/scripts/t3000/run_t3000_demo_tests.sh (hunk 1 above)
Comment (cglagovichTT) [reply tests/scripts/t3000/run_t3000_demo_tests.sh:34]: You can assign it to me

File: models/demos/t3000/llama2_70b/tt/llama_common.py (hunk 2 above)
Comment (cglagovichTT) [reply models/demos/t3000/llama2_70b/tt/llama_common.py]: I fixed this in a commit and squashed everything into two commits

**Review by caixunshiren — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

File: tests/scripts/t3000/run_t3000_demo_tests.sh (hunk 1 above)
Comment (cglagovichTT) [reply tests/scripts/t3000/run_t3000_demo_tests.sh:34]: @ttmchiou is this alright to merge?

**Review by ttmchiou — APPROVED** [review]

File: tests/scripts/t3000/run_t3000_demo_tests.sh (hunk 1 above)
Comment (ttmchiou) [reply tests/scripts/t3000/run_t3000_demo_tests.sh:34]: https://github.com/tenstorrent/tt-metal/issues/10625
for reference

**Pull request merged**